src/libs/ajax/file-browser-providers/GCSFileBrowserProvider.test.ts

```diff
@@ -0,0 +1,134 @@
+import { Ajax } from 'src/libs/ajax'
+import { FileBrowserDirectory, FileBrowserFile } from 'src/libs/ajax/file-browser-providers/FileBrowserProvider'
+import GCSFileBrowserProvider from 'src/libs/ajax/file-browser-providers/GCSFileBrowserProvider'
+import { GCSItem, GoogleStorageMethods } from 'src/libs/ajax/GoogleStorage'
+import * as Utils from 'src/libs/utils'
+import { asMockedFn } from 'src/test-utils'
+
+
+jest.mock('src/libs/ajax')
+
+const gcsObject = (name: string): GCSItem => ({
+  bucket: 'test-bucket',
+  crc32c: 'crc32c',
+  etag: 'etag',
+  generation: '1666792590000000',
+  id: `test-bucket}/${name}/1666792590000000`,
+  kind: 'storage#object',
+  md5Hash: 'abcdef',
+  mediaLink: `https://storage.googleapis.com/download/storage/v1/b/test-bucket/o/${name}?generation=1666792590000000&alt=media`,
+  metageneration: '1',
+  name,
+  selfLink: `https://www.googleapis.com/storage/v1/b/test-bucket/o/${name}`,
+  size: '1',
+  storageClass: 'STANDARD',
+  timeCreated: '2022-10-26T13:56:30.000Z',
+  timeStorageClassUpdated: '2022-10-26T13:56:30.000Z',
+  updated: '2022-10-26T13:56:30.000Z'
+})
+
+const expectedFile = (path: string): FileBrowserFile => ({
+  path,
+  url: `gs://test-bucket/${path}`,
+  size: 1,
+  createdAt: 1666792590000,
+  updatedAt: 1666792590000
+})
+
+describe('GCSFileBrowserProvider', () => {
+  let list
+
+  beforeEach(() => {
+    list = jest.fn().mockImplementation((_googleProject, _bucket, _prefix, options = {}) => {
+      const { pageToken } = options
+
+      const response: Awaited<ReturnType<GoogleStorageMethods['list']>> = Utils.switchCase(pageToken,
```

Comment:
Specifying the type currently isn't doing much good, since `switchCase` has no types and thus returns any.